src/storage/src/source/kafka.rs

```diff
@@ -948,6 +949,32 @@ fn construct_source_message(
                     .into();
                 packer.push(d)
             }
+            KafkaMetadataKind::Header { key, use_bytes } => {
+                match msg.headers() {
+                    Some(headers) => {
+                        let d = headers
+                            .iter()
+                            .find(|header| header.key == key)
+                            .map(|header| match header.value {
+                                Some(v) => {
+                                    if *use_bytes {
+                                        Datum::Bytes(v)
+                                    } else {
+                                        match str::from_utf8(v) {
+                                            Ok(str) => Datum::String(str),
+                                            Err(_) => Datum::Null,
```

Comment:
This will silently drop headers that don't parse as utf8. I'm not sure how to best surface this. Maybe at least add a debug message to the internal logs?

Reply:
This should turn the whole row into an error where the error contains the original bytes. This means this function needs to change so that it returns a `Result`